### PR TITLE
daemon: start splitting snaps op tests out of api_test.go

### PR DESCRIPTION
**Pull request by pedronis**
this splits /v2/snaps GET and POST tests out of api_test.go
to api_snaps_test.go and api_sideload_n_try_test.go

api.go itself isn't split yet similarly but for clean test writing
two changes were made

* introduction of snapInstruction.dispatchForMany used in postSnaps
  as well (with a new test about the unsupported ops case)
* changed trySnap signature

some imports in api_snaps_test.go are commented out atm,
they were copied from api_test.go and most of them will likely
come back soon

### CODE REVIEW

File: daemon/api.go [hunk 1]
```diff
@@ -732,6 +732,7 @@ func snapshotMany(inst *snapInstruction, st *state.State) (*snapInstructionResul
 }
 
 type snapActionFunc func(*snapInstruction, *state.State) (string, []*state.TaskSet, error)
+type snapsActionFunc func(*snapInstruction, *state.State) (*snapInstructionResult, error)
```
Comment (bboozzoo) [inline daemon/api.go:735]: could this be named `manySnapsActionFunc` to distinguish it visually, and also would fit nicely with `dispatchMany() (op manySnapsActionFunc)` ?

**Review by bboozzoo — APPROVED** [review]
Thank you

File: daemon/api.go (hunk 1 above)
Comment (pedronis) [reply daemon/api.go:735]: to be clear I picked the name to follow postSnap vs postSnaps (which maybe is problematic in itself).  I suppose we could as well rename postSnaps to postSnapMany?

Comment (bboozzoo) [reply daemon/api.go:735]: Yeah, that would be nice too.

Comment (pedronis) [reply daemon/api.go:735]: ok, I think I'll do both in a the follow-up though, where I will move current postSnaps to api_snaps.go

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**